server/WebApiTemplate/Controllers/UsersController.cs

```diff
@@ -0,0 +1,88 @@
+using Microsoft.AspNetCore.Mvc;
+using System.Threading;
+
+namespace EmeraldChameleonChat.Controllers
+{
+    [Route("api/[controller]/[action]")]
+    [AllowAnonymous]
+    [ApiController]
+    public class UsersController : ControllerBase
+    {
+        private readonly IUserService _userService;
+        private readonly IAuthenticationService _authenticationService;
+
+        public UsersController(IUserService userService, IAuthenticationService authenticationService)
+        {
+            _userService = userService;
+            _authenticationService = authenticationService;
+        }
+
+
+        [HttpPost(Name = "Login")]
+        public async Task<ActionResult<LoginResponseModel>> Login([FromForm] LoginModel loginDTO, CancellationToken cancellationToken)
+        {
+            if (loginDTO == null) return BadRequest("Please fill the required data");
+            if (loginDTO.Password == null) return BadRequest("Password is Required");
+            if (loginDTO.Username == null && loginDTO.Email == null) return BadRequest("Username or Email is Required");
+
+            var user = await _userService.FindUser(loginDTO);
+            if (user == null) return BadRequest("User with this username or email does not exist");
+
+            var token = await _userService.LoginUser(user, loginDTO.Password, cancellationToken);
+
+            return Ok(token);
+
+        }
+
+        [HttpPost(Name = "Register")]
+        public async Task<ActionResult<LoginResponseModel>> Register([FromForm] RegisterModel registeredUser, CancellationToken cancellationToken)
+        {
+            if (registeredUser == null) return BadRequest("Please fill the required data");
+            if (registeredUser.Username == null) return BadRequest("Username is Required");
+            if (registeredUser.Email == null) return BadRequest("Email is Required");
+            if (registeredUser.Password == null) return BadRequest("Password is Required");
+
+
+            var token = await _userService.RegisterUser(registeredUser, cancellationToken);
+            return Ok(token);
+        }
+
+        [HttpPost(Name = "SendConfirmation")]
+        public async Task<IActionResult> SendConfirmation([FromForm] string email)
+        {
+            var user = await _userService.FindUser(new LoginModel() { Email = email});
+            if (user == null) return BadRequest("User with this email does not exist");
+
+            await _userService.SendConfirmation(user);
+
+            return Ok();
+        }
+        [HttpGet(Name = "Confirmation")]
+        public async Task<IActionResult> Confirmation([FromQuery] string confirmationCode, [FromQuery] string userName, CancellationToken cancellationToken)
+        {
+            var isConfirmed = await _userService.ConfirmUser(userName, confirmationCode, cancellationToken);
+            if (isConfirmed == false) return BadRequest();
+
+            return Ok("User Confirmed");
+        }
+
+        [HttpGet(Name = "Refresh")]
```

Comment:
I prefer being verbose when naming endpoints(or anything really). I think this should become /RefreshSession & the endpoint above should be /EmailConfirmation